src/file_tree.ml

```diff
@@ -54,8 +54,21 @@ let load ?(extra_ignored_subtrees=Path.Set.empty) path =
     let files = String_set.of_list files in
     let ignored_sub_dirs =
       if not ignored && String_set.mem "jbuild-ignore" files then
-        String_set.of_list
-          (Io.lines_of_file (Path.to_string (Path.relative path "jbuild-ignore")))
+        let ignore_file = Path.to_string (Path.relative path "jbuild-ignore") in
+        let files =
+          Io.lines_of_file ignore_file
+        in
+        let remove_subdirs (line, valid) fn =
+          let line = succ line in
+          if Filename.dirname fn <> Filename.current_dir_name then begin
+            Format.eprintf "@{<loc>File \"%s\", line %d:@}@\n\
+                            @{<warning>Warning@}: subdirectory expression %s ignored@."
```

Comment:
Can you use `Loc.warn`?

Reply:
Done - having to convert it to a `Loc.t` is a little tedious, but better not to be copying the format strings.